modules/storages/lib/open_project/storages/engine.rb

```diff
@@ -135,6 +135,21 @@ def self.external_file_permissions
         ) do |payload|
           ::Storages::HealthService.new(storage: payload[:storage]).healthy
         end
+
+        OpenProject::Notifications.subscribe(
+          ::OpenIDConnect::UserTokens::FetchService::TOKEN_OBTAINED
+        ) do |payload|
+          audience = payload[:audience]
+          token = payload[:token]
+          storage = Storages::Storage.all.find { |s| s.audience == audience }
+          if storage
+            RemoteIdentities::CreateService
+              .call(user: token.user, integration: storage, token:)
+              .on_failure { raise "RemoteIdentity creation failed" }
+          else
+            puts "WARNING: no integration was found for audience: #{audience}"
```

Comment:
:green_circle: I'd leave this debug log out to be honest. Going forward it will be normal to see tokens being obtained for things that are not storages. We shouldn't expect that every exchange happens for a storage.